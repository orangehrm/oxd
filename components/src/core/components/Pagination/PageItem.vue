<template>
  <li>
    <button type="button" :class="classes">
      <oxd-icon v-if="previous" name="chevron-left" />
      <oxd-icon v-else-if="next" name="chevron-right" />
      <template v-else>{{ page }}</template>
    </button>
  </li>
</template>

<script lang="ts">
import {defineComponent} from 'vue';
import Icon from '@orangehrm/oxd/core/components/Icon/Icon.vue';
import {pageableMixin} from '../../../mixins/pageable';

export default defineComponent({
  name: 'oxd-pagination-page-item',

  components: {
    'oxd-icon': Icon,
  },

  mixins: [pageableMixin],

  props: {
    previous: {
      type: Boolean,
    },
    next: {
      type: Boolean,
    },
    selected: {
      type: Boolean,
    },
  },

  computed: {
    classes(): object {
      return {
        'oxd-pagination-page-item': true,
        'oxd-pagination-page-item--page': !(this.previous || this.next),
        'oxd-pagination-page-item--page-selected': this.selected,
        'oxd-pagination-page-item--previous-next': this.previous || this.next,
      };
    },
  },
});
</script>

<style src="./page-item.scss" lang="scss" scoped></style>
