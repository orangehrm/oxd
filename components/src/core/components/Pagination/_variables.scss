$oxd-pagination-page-item-border: $oxd-border !default;
$oxd-pagination-page-item-border-color: $oxd-interface-gray-color !default;
$oxd-pagination-page-item-margin: 0.3rem !default;
$oxd-pagination-page-item-font-size: 12px !default;
$oxd-pagination-page-item-font-family: $oxd-font-family !default;
$oxd-pagination-page-item-font-weight: 600 !default;
$oxd-pagination-page-item-font-color: $oxd-interface-gray-color !default;
$oxd-pagination-page-item-size: 40px !default;
$oxd-pagination-page-item--page-selected-bg-color: $oxd-background-light-gray-color !default;
$oxd-pagination-page-item--page-hover-font-color: $oxd-white-color !default;
$oxd-pagination-page-item--page-hover-bg-color: $oxd-primary-one-color !default;
$oxd-pagination-page-item--previous-next-hover-bg-color: $oxd-background-light-gray-color !default;
