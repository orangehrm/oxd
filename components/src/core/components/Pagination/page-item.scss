@import '../../../styles';
@import 'variables';

.oxd-pagination-page-item {
  font-size: $oxd-pagination-page-item-font-size;
  font-family: $oxd-pagination-page-item-font-family;
  font-weight: $oxd-pagination-page-item-font-weight;
  background-color: $oxd-white-color;
  color: $oxd-pagination-page-item-font-color;
  border-radius: 50%;
  cursor: pointer;
  height: $oxd-pagination-page-item-size;
  width: $oxd-pagination-page-item-size;
  margin: $oxd-pagination-page-item-margin;
  display: flex;
  justify-content: center;
  flex-direction: column;

  // TODO: handle focus
  // &:focus {
  //   outline: 0;
  // }

  &--previous-next {
    border: $oxd-pagination-page-item-border
      $oxd-pagination-page-item-border-color;

    // TODO: handle focus
    // &:focus,
    &:hover {
      background-color: $oxd-pagination-page-item--previous-next-hover-bg-color;
    }
  }

  &--page {
    border: 0;

    &-selected {
      background-color: $oxd-pagination-page-item--page-selected-bg-color;
    }

    // TODO: handle focus
    // &:focus,
    &:hover {
      color: $oxd-pagination-page-item--page-hover-font-color;
      background-color: $oxd-pagination-page-item--page-hover-bg-color;
    }
  }
}
