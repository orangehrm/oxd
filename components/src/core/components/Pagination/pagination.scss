@import '../../../styles';
@import 'variables';

.oxd-pagination__ul {
  list-style-type: none;
  display: inline-flex;
  margin: 0;
  padding: 0;
}
