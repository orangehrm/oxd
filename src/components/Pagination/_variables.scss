// This file is part of OrangeHRM Inc
//
// Copyright (C) 2020 onwards OrangeHRM Inc
//
// This program is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see  http://www.gnu.org/licenses

$oxd-pagination-page-item-border: $oxd-border !default;
$oxd-pagination-page-item-border-color: $oxd-interface-gray-color !default;
$oxd-pagination-page-item-margin: 0.3rem !default;
$oxd-pagination-page-item-font-size: 12px !default;
$oxd-pagination-page-item-font-family: $oxd-font-family !default;
$oxd-pagination-page-item-font-weight: 600 !default;
$oxd-pagination-page-item-bg-color: $oxd-white-color !default;
$oxd-pagination-page-item-font-color: $oxd-interface-gray-color !default;
$oxd-pagination-page-item-size: 36px !default;
$oxd-pagination-page-item--page-selected-bg-color: $oxd-background-light-gray-color !default;
$oxd-pagination-page-item--page-hover-font-color: $oxd-primary-font-color !default;
$oxd-pagination-page-item--page-hover-bg-color: $oxd-primary-one-color !default;
$oxd-pagination-page-item--previous-next-hover-bg-color: $oxd-background-light-gray-color !default;
