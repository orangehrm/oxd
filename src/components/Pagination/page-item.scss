// This file is part of OrangeHRM Inc
//
// Copyright (C) 2020 onwards OrangeHRM Inc
//
// This program is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see  http://www.gnu.org/licenses

@import '../../styles';
@import 'variables';

.oxd-pagination-page-item {
  font-size: $oxd-pagination-page-item-font-size;
  font-family: $oxd-pagination-page-item-font-family;
  font-weight: $oxd-pagination-page-item-font-weight;
  background-color: $oxd-pagination-page-item-bg-color;
  color: $oxd-pagination-page-item-font-color;
  border-radius: 50%;
  cursor: pointer;
  height: $oxd-pagination-page-item-size;
  width: $oxd-pagination-page-item-size;
  margin: $oxd-pagination-page-item-margin;

  /* Remove the focus indicator on mouse-focus
     https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible */
  &:focus:not(:focus-visible) {
    outline: 0;
  }

  &:focus-visible {
    outline: auto;
  }

  &--previous-next {
    border: $oxd-pagination-page-item-border
      $oxd-pagination-page-item-border-color;

    &:hover {
      background-color: $oxd-pagination-page-item--previous-next-hover-bg-color;
    }
  }

  &--page {
    border: 0;

    &-selected {
      background-color: $oxd-pagination-page-item--page-selected-bg-color;
    }

    &:hover {
      color: $oxd-pagination-page-item--page-hover-font-color;
      background-color: $oxd-pagination-page-item--page-hover-bg-color;
    }
  }
}
