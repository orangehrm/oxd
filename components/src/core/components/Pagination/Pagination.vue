<template>
  <nav role="navigation" aria-label="Pagination Navigation">
    <ul class="oxd-pagination__ul">
      <oxd-pagination-page-item
        previous
        @click="onClickPrevious"
        v-if="showPrevious"
      />
      <oxd-pagination-page-item
        v-for="page in pageItems"
        :key="page"
        :page="page"
        :selected="page === currentPage"
        @click="onClickPage(page, $event)"
      />
      <oxd-pagination-page-item next @click="onClickNext" v-if="showNext" />
      <oxd-select-input
        class="pagination-dropdown"
        :options="perPages"
        @update:modelValue="selectPerPage"
        :model-value="perPageData"
        :hide-dropdown-default-label="true"
      />
    </ul>
  </nav>
</template>

<script lang="ts">
import {defineComponent, PropType} from 'vue';
import PageItem from '@orangehrm/oxd/core/components/Pagination/PageItem.vue';
import SelectInput from '@orangehrm/oxd/core/components/Input/Select/SelectInput.vue';
import {pageableMixin} from '../../../mixins/pageable';

export default defineComponent({
  name: 'oxd-pagination',

  components: {
    'oxd-pagination-page-item': PageItem,
    'oxd-select-input': SelectInput,
  },

  mixins: [pageableMixin],

  emits: ['previous', 'next', 'update:current', 'clickPage'],

  data() {
    return {
      pagePointer: this.current,
      perPageData: {
        id: 1,
        label: '10',
      },
    };
  },

  props: {
    length: {
      type: Number,
      required: true,
      validator: (val: number) => Number.isInteger(val),
    },
    max: {
      type: Number,
      default: 5,
      validator: (val: number) => Number.isInteger(val),
    },
    current: {
      type: Number,
      default: 1,
      validator: (val: number) => Number.isInteger(val),
    },
    pagesList: {
      type: Array,
      default: () => ['10', '20', '50', '100'],
    },
    perPage: {
      type: Object as PropType<{
        id: number;
        label: string;
      }>,
      default: () => ({
        id: 1,
        label: '10',
      }),
    },
  },

  computed: {
    perPages() {
      return this.pagesList.map((page, index) => {
        return {
          id: ++index,
          label: page,
        };
      });
    },
    currentPage: {
      get(): number {
        if (this.current < 1 || this.current > this.length) {
          // eslint-disable-next-line no-console
          console.error('Invalid `current` prop');
        }
        return this.pagePointer;
      },
      set(newValue: number) {
        if (newValue < 1 || newValue > this.length) {
          // eslint-disable-next-line no-console
          console.error('Invalid assignment to `currentPage`');
        } else {
          this.pagePointer = newValue;
        }
      },
    },
    showPrevious(): boolean {
      return this.currentPage !== 1;
    },
    showNext(): boolean {
      return this.currentPage !== this.length;
    },
    pageItems(): Array<number> {
      if (this.currentPage < 1 || this.currentPage > this.length) {
        // eslint-disable-next-line no-console
        console.error('Invalid `current` prop');
      }

      const maxLength = this.max;
      const currentPage = this.currentPage;
      if (this.length <= maxLength) {
        return this.range(1, this.length);
      }

      const even = maxLength % 2 === 0 ? 1 : 0;
      const middle = Math.floor(maxLength / 2);
      let start = currentPage - middle;
      let end = currentPage + middle - even;

      if (start < 0) {
        start = 0;
      }
      if (end > this.length) {
        end = this.length;
      }
      if (currentPage <= middle) {
        end = maxLength;
      }
      if (currentPage > this.length - middle) {
        start = this.length - maxLength + 1;
      }

      return this.range(start, end);
    },
  },

  methods: {
    onClickPrevious(e: Event) {
      this.currentPage--;
      this.$emit('update:current', this.currentPage);
      this.$emit('previous', e);
    },
    onClickPage(page: number, e: Event) {
      this.currentPage = page;
      this.$emit('update:current', this.currentPage);
      this.$emit('clickPage', {
        page,
        native: e,
      });
    },
    onClickNext(e: Event) {
      this.currentPage++;
      this.$emit('update:current', this.currentPage);
      this.$emit('next', e);
    },
    range(from: number, to: number): Array<number> {
      const range = [];
      from = from > 0 ? from : 1;
      if (from > to) {
        // eslint-disable-next-line no-console
        console.error('`from` is bigger than `to`');
      }
      for (let i = from; i <= to; i++) {
        range.push(i);
      }
      return range;
    },
    selectPerPage(val) {
      this.perPageData = val;
      this.$emit('onPerPageSelect', val);
    },
  },

  mounted() {
    this.perPageData = this.perPage;
  },
});
</script>

<style src="./pagination.scss" lang="scss" scoped></style>

<style lang="scss" scoped>
:deep(.pagination-dropdown .oxd-select-text-input) {
  min-width: 59px;
}
</style>
