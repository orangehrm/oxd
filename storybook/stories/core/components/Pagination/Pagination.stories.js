import Pagination from '@orangehrm/oxd/core/components/Pagination/Pagination';

export default {
  title: 'Example/Pagination',
  component: Pagination,
};

const Template = (args, {argTypes}) => ({
  props: Object.keys(argTypes),
  data() {
    return {
      current: 2,
    };
  },
  components: {'oxd-pagination': Pagination},
  template:
    '<div>Current page: {{current}} </div><oxd-pagination :length="length" v-model:current="current" :max="max" />',
});

export const Default = Template.bind({});
Default.args = {
  length: 3,
  max: 6,
};

// export const Colored = Template.bind({});
// Colored.args = {
//   value: 'Textarea',
//   style: {backgroundColor: 'aliceblue'},
// };

// export const Error = Template.bind({});
// Error.args = {
//   value: 'Textarea',
//   hasError: true,
// };
