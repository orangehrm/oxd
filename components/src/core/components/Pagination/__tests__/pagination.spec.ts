import {mount} from '@vue/test-utils';
import Pagination from '@orangehrm/oxd/core/components/Pagination/Pagination.vue';

describe('Pagination.vue', () => {
  it('renders OXD Pagination', () => {
    const wrapper = mount(Pagination, {
      props: {length: 6},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Pagination with length < max', () => {
    const wrapper = mount(Pagination, {
      props: {length: 3, max: 5},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Pagination with length > max', () => {
    const wrapper = mount(Pagination, {
      props: {length: 21, max: 6},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Pagination with current', () => {
    const wrapper = mount(Pagination, {
      props: {length: 21, current: 21},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Pagination with |1|, 2, 3, 4, 5', () => {
    const wrapper = mount(Pagination, {
      props: {length: 21, current: 1, max: 5},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Pagination with 1, |2|, 3, 4, 5', () => {
    const wrapper = mount(Pagination, {
      props: {length: 21, current: 2, max: 5},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Pagination with 1, 2, |3|, 4, 5', () => {
    const wrapper = mount(Pagination, {
      props: {length: 21, current: 3, max: 5},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Pagination with 1, 2, 3, |4|, 5, 6', () => {
    const wrapper = mount(Pagination, {
      props: {length: 21, current: 4, max: 6},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Pagination with |17|, 18, 19, 20, 21', () => {
    const wrapper = mount(Pagination, {
      props: {length: 21, current: 17, max: 5},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Pagination with 17, |18|, 19, 20, 21', () => {
    const wrapper = mount(Pagination, {
      props: {length: 21, current: 18, max: 5},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Pagination with 17, 18, |19|, 20, 21', () => {
    const wrapper = mount(Pagination, {
      props: {length: 21, current: 19, max: 5},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Pagination with 17, 18, 19, |20|, 21', () => {
    const wrapper = mount(Pagination, {
      props: {length: 21, current: 20, max: 5},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Pagination with 17, 18, 19, 20, |21|', () => {
    const wrapper = mount(Pagination, {
      props: {length: 21, current: 21, max: 5},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
