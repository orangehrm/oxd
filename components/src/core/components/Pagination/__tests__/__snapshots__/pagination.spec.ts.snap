// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination.vue renders OXD Pagination 1`] = `
<nav role="navigation" aria-label="Pagination Navigation">
  <ul class="oxd-pagination__ul">
    <!--v-if-->
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page oxd-pagination-page-item--page-selected">1</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">2</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">3</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">4</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">5</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--previous-next"><i class="oxd-icon bi-chevron-right"></i></button></li>
  </ul>
</nav>
`;

exports[`Pagination.vue should renders OXD Pagination with |1|, 2, 3, 4, 5 1`] = `
<nav role="navigation" aria-label="Pagination Navigation">
  <ul class="oxd-pagination__ul">
    <!--v-if-->
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page oxd-pagination-page-item--page-selected">1</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">2</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">3</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">4</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">5</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--previous-next"><i class="oxd-icon bi-chevron-right"></i></button></li>
  </ul>
</nav>
`;

exports[`Pagination.vue should renders OXD Pagination with |17|, 18, 19, 20, 21 1`] = `
<nav role="navigation" aria-label="Pagination Navigation">
  <ul class="oxd-pagination__ul">
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--previous-next"><i class="oxd-icon bi-chevron-left"></i></button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">15</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">16</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page oxd-pagination-page-item--page-selected">17</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">18</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">19</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--previous-next"><i class="oxd-icon bi-chevron-right"></i></button></li>
  </ul>
</nav>
`;

exports[`Pagination.vue should renders OXD Pagination with 1, |2|, 3, 4, 5 1`] = `
<nav role="navigation" aria-label="Pagination Navigation">
  <ul class="oxd-pagination__ul">
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--previous-next"><i class="oxd-icon bi-chevron-left"></i></button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">1</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page oxd-pagination-page-item--page-selected">2</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">3</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">4</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">5</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--previous-next"><i class="oxd-icon bi-chevron-right"></i></button></li>
  </ul>
</nav>
`;

exports[`Pagination.vue should renders OXD Pagination with 1, 2, |3|, 4, 5 1`] = `
<nav role="navigation" aria-label="Pagination Navigation">
  <ul class="oxd-pagination__ul">
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--previous-next"><i class="oxd-icon bi-chevron-left"></i></button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">1</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">2</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page oxd-pagination-page-item--page-selected">3</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">4</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">5</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--previous-next"><i class="oxd-icon bi-chevron-right"></i></button></li>
  </ul>
</nav>
`;

exports[`Pagination.vue should renders OXD Pagination with 1, 2, 3, |4|, 5, 6 1`] = `
<nav role="navigation" aria-label="Pagination Navigation">
  <ul class="oxd-pagination__ul">
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--previous-next"><i class="oxd-icon bi-chevron-left"></i></button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">1</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">2</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">3</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page oxd-pagination-page-item--page-selected">4</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">5</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">6</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--previous-next"><i class="oxd-icon bi-chevron-right"></i></button></li>
  </ul>
</nav>
`;

exports[`Pagination.vue should renders OXD Pagination with 17, |18|, 19, 20, 21 1`] = `
<nav role="navigation" aria-label="Pagination Navigation">
  <ul class="oxd-pagination__ul">
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--previous-next"><i class="oxd-icon bi-chevron-left"></i></button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">16</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">17</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page oxd-pagination-page-item--page-selected">18</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">19</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">20</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--previous-next"><i class="oxd-icon bi-chevron-right"></i></button></li>
  </ul>
</nav>
`;

exports[`Pagination.vue should renders OXD Pagination with 17, 18, |19|, 20, 21 1`] = `
<nav role="navigation" aria-label="Pagination Navigation">
  <ul class="oxd-pagination__ul">
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--previous-next"><i class="oxd-icon bi-chevron-left"></i></button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">17</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">18</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page oxd-pagination-page-item--page-selected">19</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">20</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">21</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--previous-next"><i class="oxd-icon bi-chevron-right"></i></button></li>
  </ul>
</nav>
`;

exports[`Pagination.vue should renders OXD Pagination with 17, 18, 19, |20|, 21 1`] = `
<nav role="navigation" aria-label="Pagination Navigation">
  <ul class="oxd-pagination__ul">
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--previous-next"><i class="oxd-icon bi-chevron-left"></i></button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">17</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">18</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">19</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page oxd-pagination-page-item--page-selected">20</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">21</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--previous-next"><i class="oxd-icon bi-chevron-right"></i></button></li>
  </ul>
</nav>
`;

exports[`Pagination.vue should renders OXD Pagination with 17, 18, 19, 20, |21| 1`] = `
<nav role="navigation" aria-label="Pagination Navigation">
  <ul class="oxd-pagination__ul">
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--previous-next"><i class="oxd-icon bi-chevron-left"></i></button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">17</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">18</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">19</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">20</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page oxd-pagination-page-item--page-selected">21</button></li>
    <!--v-if-->
  </ul>
</nav>
`;

exports[`Pagination.vue should renders OXD Pagination with current 1`] = `
<nav role="navigation" aria-label="Pagination Navigation">
  <ul class="oxd-pagination__ul">
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--previous-next"><i class="oxd-icon bi-chevron-left"></i></button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">17</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">18</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">19</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">20</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page oxd-pagination-page-item--page-selected">21</button></li>
    <!--v-if-->
  </ul>
</nav>
`;

exports[`Pagination.vue should renders OXD Pagination with length < max 1`] = `
<nav role="navigation" aria-label="Pagination Navigation">
  <ul class="oxd-pagination__ul">
    <!--v-if-->
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page oxd-pagination-page-item--page-selected">1</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">2</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">3</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--previous-next"><i class="oxd-icon bi-chevron-right"></i></button></li>
  </ul>
</nav>
`;

exports[`Pagination.vue should renders OXD Pagination with length > max 1`] = `
<nav role="navigation" aria-label="Pagination Navigation">
  <ul class="oxd-pagination__ul">
    <!--v-if-->
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page oxd-pagination-page-item--page-selected">1</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">2</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">3</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">4</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">5</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--page">6</button></li>
    <li><button type="button" class="oxd-pagination-page-item oxd-pagination-page-item--previous-next"><i class="oxd-icon bi-chevron-right"></i></button></li>
  </ul>
</nav>
`;
